src/Features/CSharp/Portable/DocumentSymbols/CSharpDocumentSymbolsService.cs

```diff
@@ -0,0 +1,248 @@
+                        or AttributeListSyntax or TypeParameterConstraintClauseSyntax;
+        }
+
+        protected override DocumentSymbolInfo GetMemberInfoForType(INamedTypeSymbol type, SyntaxTree tree, ISymbolDeclarationService declarationService, CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<DocumentSymbolInfo>.GetInstance(out var membersBuilder);
+
+            foreach (var member in type.GetMembers())
+            {
+                switch (member)
+                {
+                    case { IsImplicitlyDeclared: true } or { Kind: SymbolKind.NamedType } or
+                         IMethodSymbol { MethodKind: MethodKind.PropertyGet or MethodKind.PropertySet }:
+                        continue;
+
+                    case IMethodSymbol { PartialImplementationPart: { } partialPart }:
+                        membersBuilder.Add(CreateInfo(member, tree, declarationService, ImmutableArray<DocumentSymbolInfo>.Empty, cancellationToken));
+                        membersBuilder.Add(CreateInfo(partialPart, tree, declarationService, ImmutableArray<DocumentSymbolInfo>.Empty, cancellationToken));
+                        break;
+
+                    default:
+                        membersBuilder.Add(CreateInfo(member, tree, declarationService, ImmutableArray<DocumentSymbolInfo>.Empty, cancellationToken));
+                        break;
+                }
+            }
+
+            membersBuilder.Sort((d1, d2) => d1.Text.CompareTo(d2.Text));
+            return CreateInfo(type, tree, declarationService, membersBuilder.ToImmutable(), cancellationToken);
+        }
+
+        protected override DocumentSymbolInfo CreateInfo(
+            ISymbol symbol,
+            SyntaxTree tree,
+            ISymbolDeclarationService declarationService,
+            ImmutableArray<DocumentSymbolInfo> childrenSymbols,
+            CancellationToken cancellationToken)
+        {
+            var enclosingSpans = GetEnclosingSpansForSymbol(symbol, tree, declarationService);
+            var declaringSpans = GetDeclaringSpans(symbol, tree);
+            var obsolete = symbol.GetAttributes().Any(attr => attr.AttributeClass?.MetadataName == "ObsoleteAttribute");
+
+            var isConstant = symbol is IFieldSymbol { IsConst: true } or ILocalSymbol { IsConst: true };
+
+            return new(
+                symbol.ToDisplayString(symbol is ITypeSymbol ? s_typeFormat : s_memberFormat),
+                symbol.Name,
+                symbol.GetGlyph(),
+                obsolete,
+                isConstant ? s_constantTag : ImmutableArray<string>.Empty,
+                DocumentSymbolInfoExtensions.GetDocumentSymbolInfoPropertiesForSymbol(symbol, cancellationToken),
+                enclosingSpans,
+                declaringSpans,
+                childrenSymbols);
+        }
+
+        protected override bool ConsiderNestedNodesChildren(ISymbol node) => node is not (ILabelSymbol or ILocalSymbol);
```

Comment:
```suggestion
        protected override bool ConsiderNestedNodesChildren(ISymbol node)
            => node is not (ILabelSymbol or ILocalSymbol);
```

mehtods with expr-bodies have the body on the next line, indented.